src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -2975,3 +2980,368 @@ void ConptyRoundtripTests::NewLinesAtBottomWithBackground()
+    // found during debugging.
+
+    auto& g = ServiceLocator::LocateGlobals();
+    auto& renderer = *g.pRender;
+    auto& gci = g.getConsoleInformation();
+    auto& si = gci.GetActiveOutputBuffer();
+    auto& sm = si.GetStateMachine();
+    auto* hostTb = &si.GetTextBuffer();
+    auto* termTb = term->_buffer.get();
+
+    _flushFirstFrame();
+
+    _checkConptyOutput = false;
+    _logConpty = true;
+
+    const size_t width = static_cast<size_t>(TerminalViewWidth);
+
+    const auto charsInFirstLine = width;
+    const auto numCharsFirstColor = 30;
+    const auto numCharsSecondColor = charsInFirstLine - numCharsFirstColor;
+    const auto charsInSecondLine = width / 2;
+
+    const auto defaultAttrs = TextAttribute();
+    const auto conhostDefaultAttrs = si.GetAttributes();
+
+    // This is a simple helper for calling WriteCharsLegacy with the correct
+    // args. It will call WCL a number of times for the provided length, writing
+    // `length` '~' characters to the buffer.
+    auto writeCharsLegacyHelper = [&](const auto length) {
+        SetVerifyOutput settings(VerifyOutputSettings::LogOnlyFailures);
+        wchar_t* str = L"~";
+        size_t seqCb = 2;
+        for (auto i = 0; i < length; i++)
+        {
+            VERIFY_SUCCESS_NTSTATUS(WriteCharsLegacy(si, str, str, str, &seqCb, nullptr, hostTb->GetCursor().GetPosition().X, writeCharsLegacyMode, nullptr));
+        }
+    };
+
+    for (auto i = 0; i < (TerminalViewHeight + circledRows) / 2; i++)
+    {
+        Log::Comment(NoThrowString().Format(L"writing pair of lines %d", i));
+
+        if (i > 0)
+        {
+            sm.ProcessString(L"\r\n");
+        }
+
+        sm.ProcessString(L"\x1b[33m");
+        if (writingMethod == PrintWithPrintString)
+        {
+            sm.ProcessString(std::wstring(numCharsFirstColor, L'~'));
+        }
+        else if (writingMethod == PrintWithWriteCharsLegacy)
+        {
+            writeCharsLegacyHelper(numCharsFirstColor);
+        }
+        sm.ProcessString(L"\x1b[m");
+
+        if (writingMethod == PrintWithPrintString)
+        {
+            sm.ProcessString(std::wstring(numCharsSecondColor, L'~'));
+        }
+        else if (writingMethod == PrintWithWriteCharsLegacy)
+        {
+            writeCharsLegacyHelper(numCharsSecondColor);
+        }
+
+        if (paintEachNewline == PaintEveryLine)
+        {
+            VERIFY_SUCCEEDED(renderer.PaintFrame());
+        }
+
+        if (writingMethod == PrintWithPrintString)
+        {
+            sm.ProcessString(std::wstring(charsInSecondLine, L'~'));
+        }
+        else if (writingMethod == PrintWithWriteCharsLegacy)
+        {
+            writeCharsLegacyHelper(charsInSecondLine);
+        }
+
+        if (paintEachNewline == PaintEveryLine ||
+            paintEachNewline == PaintAfterBothLines)
+        {
+            VERIFY_SUCCEEDED(renderer.PaintFrame());
+        }
+    }
+

```

Comment:
Please add more comments explaining this. Whomever debugs this test in the future is going to have to decode the VT sequences as well as try to reason out why these modes have different behaviors. Leave as many breadcrumbs as you can.